Lombiq.NodeJs.Extensions/Lombiq.NodeJs.Extensions.csproj

```diff
@@ -54,14 +53,64 @@
-         for PackageReferences to specify that some assets should not be consumed by itself but still forwarded to
+         for PackageReferences to specify that some assets should not be consumed by itself but still be forwarded to
          projects consuming this one. -->
     <ExecNpmInstallCommand>false</ExecNpmInstallCommand>
     <ExecDotnetPostcleanCommand>false</ExecDotnetPostcleanCommand>
     <ExecDotnetPrebuildCommand>false</ExecDotnetPrebuildCommand>
   </PropertyGroup>
 
+  <!-- For ESLint to work properly during Build AND in Visual Studio, we need to put the configuration and plugin
+       packages used by our ESLint Lombiq base configuration next to the referencing .eslintrc file. -->
+  <Target Name="SetUpGlobalESLintConfiguration" BeforeTargets="BeforeResolveReferences">
+    <PropertyGroup>
+      <NodeJsExtensionsGlobalESLintConfigurationDirectory
+        Condition="'$(NodeJsExtensionsGlobalESLintConfigurationDirectory)' != ''
+      ">$(NodeJsExtensionsGlobalESLintConfigurationDirectory.TrimEnd('/\'))</NodeJsExtensionsGlobalESLintConfigurationDirectory>
+      <ConsumerConfigurationDirectory>$(MSBuildThisFileDirectory)/config/consumer</ConsumerConfigurationDirectory>
+      <ESLintPackageJsonDirectory>$(ConsumerConfigurationDirectory)</ESLintPackageJsonDirectory>
+      <IsGlobalESLintConfigurationDirectoryValid
+        Condition="
+          '$(NodeJsExtensionsGlobalESLintConfigurationDirectory)' != ''
+          AND Exists($(NodeJsExtensionsGlobalESLintConfigurationDirectory))
+          AND !$([System.IO.File]::Exists($(NodeJsExtensionsGlobalESLintConfigurationDirectory)))
+        ">true</IsGlobalESLintConfigurationDirectoryValid>
+      <ESLintGlobalConfigRelativeDir
+        Condition="
+          '$(IsGlobalESLintConfigurationDirectoryValid)' == 'true'
+        ">$([MSBuild]::MakeRelative($(ESLintPackageJsonDirectory), $(NodeJsExtensionsGlobalESLintConfigurationDirectory)))</ESLintGlobalConfigRelativeDir>
+      <ESLintNodeModules
+        Condition="'$(ESLintGlobalConfigRelativeDir)' != ''">$(ESLintGlobalConfigRelativeDir)/node_modules</ESLintNodeModules>
+    </PropertyGroup>
+    <Warning
+      Code="NE01"
+      Text="The path '$(NodeJsExtensionsGlobalESLintConfigurationDirectory)' in 'NodeJsExtensionsGlobalESLintConfigurationDirectory' is not a directory."
+      Condition="'$(NodeJsExtensionsGlobalESLintConfigurationDirectory)' != '' AND '$(IsGlobalESLintConfigurationDirectoryValid)' != 'true'" />
+    <!-- We need pnpm, so we enable it here explicitly using `corepack enable` from Node.js >=16.9. -->
+    <Exec Command="corepack enable" IgnoreExitCode="true"
+          Condition="'$(IsGlobalESLintConfigurationDirectoryValid)' == 'true'" />
+    <!-- We use the latest pnpm@6 version, which sadly cannot be expressed more succinctly.-->
+    <Exec Command="corepack prepare pnpm@6.32.24 --activate" IgnoreExitCode="true"
+          Condition="'$(IsGlobalESLintConfigurationDirectoryValid)' == 'true'" />
```

Comment:
We can require that Node version, if it's available on GitHub Actions, there's no issue installing it in TC, and team members can do that too (https://lombiq.atlassian.net/browse/ADHOC-158).

Reply:
Just because we can, doesn't mean we should.

As it currently stands, NE will be usable even for lower Node.js versions as long as `pnpm` will be available. Why would we want to break that?

Reply:
Fair enough. The Readme asks you to enable corepack though, which is not necessary.